### ISSUE TITLE
Missing softmax on the decoder output

### ISSUE THREAD

**Issue by PT-10** [hkproj/pytorch-transformer#35]
based on the attention is all you need paper, the final projection is supposed to give the probability of each token in the vocabulary, given via softmax, current implementation leaves it at logits.

Proposed change in model.py (lines 188-196) - 
```
class ProjectionLayer(nn.Module):

    def __init__(self, d_model, vocab_size) -> None:
        super().__init__()
        self.proj = nn.Linear(d_model, vocab_size)

    def forward(self, x) -> None:
        # (batch, seq_len, d_model) --> (batch, seq_len, vocab_size)
        return torch.softmax(self.proj(x), dim=-1)
```
